### PR TITLE
chore: integration test for rETL flow

### PR DESCRIPTION
**Pull request by lvrach**
# Description

An integration test for the recently added `/retl` endpoint. This new endpoint lets users also provide destination id to which they want to route their events to. Tests added to verify sources stats' behaviour has remained the same.

## Linear Ticket

[linear](https://linear.app/rudderstack/issue/PIPE-701/retl-integration-test) resolves PIPE-701

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: integration_test/retl_test/sut_test.go [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+	var (
+		httpClient = &http.Client{}
+		url        = fmt.Sprintf("%s/v1/job-status/%s?%s", s.URL, jobRunID, url.Values{
+			"task_run_id": []string{jobTaskID},
+		}.Encode()) // job_run_id
+	)
+
+	var status rsources.JobStatus
+
+	res, err := httpClient.Get(url)
```
Comment (achettyiitr) [inline integration_test/retl_test/sut_test.go:326]: Can we use default client here?
```suggestion
	res, err := http.Get(url)
```

File: integration_test/retl_test/sut_test.go [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+		t.Fail()
+		return
+	}
+
+	t.Logf("Event Sent Successfully: (%s)", body)
+}
+
+// JobStatus hits the job-status endpoint and returns the status for a sourceID, jobRunID, jobTaskID.
+// If the job is not found, the second return value is false. Any other error is logged and the test is failed.
+func (s *SUT) JobStatus(t *testing.T, sourceID, jobRunID, jobTaskID string) (rsources.JobStatus, bool) {
```
Comment (achettyiitr) [inline integration_test/retl_test/sut_test.go:316]: sourceID not being used?
```suggestion
func (s *SUT) JobStatus(t *testing.T, jobRunID, jobTaskID string) (rsources.JobStatus, bool) {
```

File: integration_test/retl_test/sut_test.go [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+	}
+	defer func() { httputil.CloseResponse(res) }()
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		t.Logf("sendEvent error: %v", err)
+		t.Fail()
+		return
+	}
+	if res.Status != "200 OK" {
```
Comment (achettyiitr) [inline integration_test/retl_test/sut_test.go:305]: Why not check the status code?
```suggestion
	if res.StatusCode != http.StatusOK {
```

**Review by achettyiitr — APPROVED** [review]

File: integration_test/retl_test/sut.go [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+	pool, err := dockertest.NewPool("")
+	require.NoError(t, err)
+
+	containersGroup, _ := errgroup.WithContext(context.TODO())
+
+	var postgresContainer *resource.PostgresResource
+	var transformerContainer *destination.TransformerResource
+
+	containersGroup.Go(func() (err error) {
+		postgresContainer, err = resource.SetupPostgres(pool, t)
```
Comment (fracasula) [inline integration_test/retl_test/sut.go:126]: I think you need to change these, it now is `postgres.Setup(pool, t)`.

**Review by fracasula — APPROVED** [review]

File: integration_test/retl_test/sut.go (hunk 4 above)
Comment (fracasula) [reply integration_test/retl_test/sut.go]: I pushed the changes @lvrach

**Pull request merged**